### PR TITLE
New user's status field

### PR DESCRIPTION
**Pull request by daisyzarzo**
relates to 
https://github.com/bespoken/dashboard/issues/1109

### CODE REVIEW

File: src/components/BillCard/Payment.tsx [hunk 1]
```diff
@@ -47,6 +48,9 @@ class SplitFormC extends React.Component<any & SplitFormCProps, any> {
 
     setStripe = async (tokenId: any) => {
         const { email, userId } = this.props.user;
+        console.log("this.props.userDetail");
```
Comment (jperata) [inline src/components/BillCard/Payment.tsx]: remove

File: src/components/BillCard/Payment.tsx [hunk 2]
```diff
@@ -47,6 +48,9 @@ class SplitFormC extends React.Component<any & SplitFormCProps, any> {
...
+        console.log(this.props.userDetail);
```
Comment (jperata) [inline src/components/BillCard/Payment.tsx]: remove

File: src/services/auth.ts [hunk 3]
```diff
@@ -189,84 +193,88 @@ namespace auth {
...
 
     export const loginWithCustomToken = async (token: string, auth: remoteservice.auth.Auth = remoteservice.defaultService().auth(), localStorage: LocalStorage = new BrowserStorage()): Promise<User> => {
         const user: remoteservice.user.User = await auth.signInWithCustomToken(token);
         const userDetails = await currentUserDetails();
         const isNewUser = !userDetails.originalCreatedAt && !isLegacyUserWithBenefics(userDetails);
+
         if (isNewUser) {
-          await updateCurrentUser({ planId: planByDefaultForNewUser });
+            const props = { planId: planByDefaultForNewUser, status: userStatus.Trial };
+            await createUserDetails(user.uid, props);
```
Comment (jperata) [inline src/services/auth.ts]: just use, source.createUserDetail(user.uid, props)

File: src/services/auth.ts [hunk 4]
```diff
@@ -309,22 +317,26 @@ namespace auth {
...
             const userWithExtraInfo = await source.getUserWithExtraInfo(currentUser.uid);
-            return new UserDetails(silentEchoToken, smAPIAccessToken, vendorID, stripeSubscribedPlanId,
-                stripeCustomerObjId, stripeSubscribedPlanName, planId, name, company, virtualDevices, userWithExtraInfo);
+            return new UserDetails(silentEchoToken, smAPIAccessToken, vendorID, status, stripeSubscribedPlanId,
+                stripeCustomerObjId, stripeSubscribedPlanName, planId, name, company, virtualDevices, userWithExtraInfo, trialEndDate);
         }
         return emptyUser;
     };
 
+    export const createUserDetails = (uid: string, props: Object): Promise<any> => {
```
Comment (jperata) [inline src/services/auth.ts]: remove this function, it doesn't seem to be used outside auth. And you just can call the source

File: src/services/auth.ts (hunk 3 above)
Comment (daisyzarzo) [reply src/services/auth.ts]: I did it so the test was easy to mock

File: src/services/auth.ts (hunk 4 above)
Comment (daisyzarzo) [reply src/services/auth.ts]: I did it so the test was easy to mock

**Pull request merged**